### PR TITLE
Fix argument annotation in `converter.parse()` signature

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
Fixes error reported in https://github.com/cuthbertLab/music21/pull/1664#issuecomment-1817500436.

Adds missing coverage.

### CODE REVIEW

File: music21/converter/__init__.py [hunk 1]
```diff
@@ -1375,16 +1376,13 @@ def parse(value: bundles.MetadataEntry | bytes | str | pathlib.Path,
         valueStr = ''
 
     if (common.isListLike(value)
-            and isinstance(value, collections.abc.Sequence)
```
Comment (jacobtylerwalls) [inline music21/converter/__init__.py:1378]: Redundant now that isListLike no longer permits sets

File: music21/converter/__init__.py [hunk 2]
```diff
@@ -1375,16 +1376,13 @@ def parse(value: bundles.MetadataEntry | bytes | str | pathlib.Path,
...
             and len(value) == 2
             and value[1] is None
             and _osCanLoad(str(value[0]))):
         # comes from corpus.search
         return parseFile(value[0], format=format, **keywords)
     elif (common.isListLike(value)
-          and isinstance(value, collections.abc.Sequence)
           and len(value) == 2
-          and isinstance(value[1], int)
```
Comment (jacobtylerwalls) [inline music21/converter/__init__.py:1387]: Was preventing the more informative error message below, allowing later `elif` blocks to match and crash.

File: music21/converter/__init__.py [hunk 3]
```diff
@@ -2072,6 +2070,21 @@ def testConversionABCWorkFromOpus(self):
...
         # s.show()
 
+    def testConversionABCWorkFromOpusWithoutKeyword(self):
+        fp = common.getSourceFilePath() / 'corpus' / 'essenFolksong' / 'testd.abc'
+
+        s = parse((str(fp), None))
+        self.assertIsInstance(s, stream.Opus)
+
+        with self.assertRaises(ConverterException):
+            parse((fp, 8))
```
Comment (mscuthbert) [inline music21/converter/__init__.py:2080]: wow... bug.

**Review by mscuthbert — APPROVED** [review]
Thank you.  Looks great.

**Pull request merged**